docs/madr/2-sqlite.md

```diff
@@ -0,0 +1,100 @@
+# Storing ANT Registry State in AO Process
+
+- Status: proposed
+- Approvers: [Dylan], [Ariel], [Phil]
+- Date: 2024-07-22
+- Authors: [Atticus]
+
+## Context and Problem Statement
+
+There are multiple options for storing state in ao Lua processes, namely thru
+the use of the Lua `table` primitive, and a special build called aos-sqlite
+which exposes the lua-sqlite3 library for using in-memory databases.
+
+Directly using Lua tables for complex data has shown a need for better tooling
+in manipulating and searching data, making the SQLite implementation a potential
+solution. However, its novelty raises concerns about reliability in larger
+products.
+
+## Decision Drivers
+
+The main drivers for this decision are:
+
+- **Scalability**
+  - The current method of using tables has scalability issues that SQLite can
+    potentially address.
+- **Performance**
+  - Improved performance in handling larger datasets with SQLite.
```

Comment:
also not necessarily true, given our schema is fairly small

Reply:
i think the only major pro for sqlite would be support for complex schemas and optizmied querying for those complex data sets